jerry-ext/debugger/debugger-tcp.c

```diff
@@ -131,6 +131,16 @@ jerryx_debugger_tcp_close (jerry_debugger_transport_header_t *header_p) /**< tcp
 
   jerryx_debugger_transport_tcp_t *tcp_p = (jerryx_debugger_transport_tcp_t *) header_p;
 
+  /**
+   * Waiting the remaining tcp data are sent, so that the tcp socket are
+   * closed by debugger client first and the debugger client won't receive
+   * socket are closed by remote error
+   */
+  for (int i = 0; i < 5; i += 1)
+  {
+    jerry_debugger_transport_sleep ();
+  }
```

Comment:
My two cents:
- Five seems like a very arbitrary magic number as the loop count to me. May or may not work on different systems. 
- `i += 1` is better as `i++` or `++i`.
- Is this really something that has to be solved here, or could the debugger client be better prepared for premature closing of the socket?

Reply:
> My two cents:
> 
> * Five seems like a very arbitrary magic number as the loop count to me. May or may not work on different systems.
> * `i += 1` is better as `i++` or `++i`.
> * Is this really something that has to be solved here, or could the debugger client be better prepared for premature closing of the socket?

Didn't figure out the best solution yet, looking for help.
I've seeing the CI randomly failure with the debuger tests, The source is here, we need waiting the debugger client close first at normal case.

Reply:
> My two cents:
> 
> * Five seems like a very arbitrary magic number as the loop count to me. May or may not work on different systems.
> * `i += 1` is better as `i++` or `++i`.
> * Is this really something that has to be solved here, or could the debugger client be better prepared for premature closing of the socket?

Here is the things that I solved:
https://github.com/jerryscript-project/jerryscript/runs/1721915835